services/telemetry/telemetry.go

```diff
@@ -234,3 +384,39 @@ func getLinuxDistribution(procVersion string) string {
 	}
 	return "unknown"
 }
+
+func (s *Service) runWaitAndRetry(ctx context.Context, req *reporter.ReportRequest) {
+	t := time.NewTimer(retryBackoff)
+	defer t.Stop()
+
+	for i := 0; i < retryCnt; i++ {
+		select {
+		case <-t.C:
+			if err := s.retry(ctx, req); err != nil {
+				s.l.Debugf("Fail to retry send telemetry request: %s, error%+v", req, err)
+				t.Reset(retryBackoff)
```

Comment:
Looks like you want a ticker there, not a timer?

Reply:
Yep, it is place for ticker.